src/c/bus-mqtt.c

```diff
@@ -73,6 +73,10 @@ static void edgex_bus_mqtt_subscribe (void *ctx, const char *topic)
   {
     iot_log_error (cinfo->lc, "mqtt: unable to subscribe to %s, error code %d", topic, rc);
   }
+  else
+  {
+    MQTTAsync_waitForCompletion(cinfo->client, opts.token, 1000);
```

Comment:
Perhaps we could consider adding a check for the return code of MQTTAsync_waitForCompletion. If the return code is not MQTTASYNC_SUCCESS, logging it could be helpful for diagnosing problems.

Reply:
@eaton-coreymutter Just wanted to confirm the idea of adding a return code check for MQTTAsync_waitForCompletion. What are your thoughts on this?

Reply:
Yes, it's a good idea, I will add the change soonish, or you are welcome to.

Reply:
@eaton-coreymutter I am asking Felix to merge this PR and to open another PR to address his change request